### ISSUE TITLE
Getting "document is not defined" with Next.js

### ISSUE THREAD

**Issue by shehzadasif** [shshaw/Splitting#80]
Hi there,

I was trying your awesome library that works seamlessly without any framework. I tried it with Next.js and no matter what I do, I am constantly getting the error as attached "document is not defined". Any idea on how to solve this issue and use it with next.js.

![splitting error](https://user-images.githubusercontent.com/96428705/147649774-e013de5c-f462-428b-8318-d2be1e28cd17.png)

**Comment by jeff-zhenz** [comment]
see first section of this https://nextjs.org/docs/advanced-features/dynamic-import, client-side work only

**Comment by shehzadasif** [comment]
Thanks for the reply, document error is resolved but I couldn't figure out
the way to get data-splitting attributes working in this way. I will really
appreciate a working demo for the same.

**Comment by TomSmedley** [comment]
I'm having this issue with Svelte. I think since everything is moving over to Vite we really need a ESM module as well as IIFE.

**Comment by d-e-v-esh** [comment]
Yes, I am having the same issue with next.js. A working solution with dynamic imports will be really appreciated.

**Comment by onopko81** [comment]
Hi, try: 
 const Splitting = null;
if (typeof document !== `undefined`) Splitting = require('Splitting')

**Comment by joshtom** [comment]
Also having some difficulties to get it to work in NextJS.

I managed to bypass the document undefined using a dynamic import using

```
 const callSplitting = async () => {
    const Splitting = (await import("splitting")).default;
    if (splitRef) {
      let split_res = Splitting();
      setLines(split_res[0].el);
      console.log(`Split text into ${split_res[0].el} lines`);
    }
  };
```

And this 

```
<div ref={splitRef} data-splitting="">
Content Here
</div>
```

But somehow, It's not just working. I would appreciate a solution regarding this @shshaw

**Comment by draZer0** [comment]
For anyone who is still struggling to use Splitting with Next.js, I combined all the answers from previous issues and I managed to get it working in with the following code:

```JSX
import "splitting/dist/splitting.css";
import "splitting/dist/splitting-cells.css";

const Component = () => {

    let target;

    setTimeout(() => {
        if ( window && document && target ) {
            const Splitting = require('Splitting');
            Splitting({ by: "chars", target: target,  });
        }
    });

    return (
        <>
            <span ref={(el) => { target = el; }}>Lorem ipsum dolor sit amet</span>
        </>
    )
}

export default Component;
```

**Comment by joshtom** [comment]
> For anyone who is still struggling to use Splitting with Next.js, I combined all the answers from previous issues and I managed to get it working in with the following code:
> 
> ```js
> import "splitting/dist/splitting.css";
> import "splitting/dist/splitting-cells.css";
> 
> const Component = () => {
> 
>     let target;
> 
>     setTimeout(() => {
>         if ( window && document && target ) {
>             const Splitting = require('Splitting');
>             Splitting({ by: "chars", target: target,  });
>         }
>     });
> 
>     return (
>         <>
>             <span ref={(el) => { target = el; }}>Lorem ipsum dolor sit amet</span>
>         </>
>     )
> }
> 
> export default Component;
> ```

Thank you for this @draZer0
